### PR TITLE
WFLY-9417 Add ability to control the startup behaviour for deployments

### PR DESCRIPTION
**Pull request by stuartwdouglas**
https://issues.jboss.org/browse/WFLY-9417

### CODE REVIEW

**Review by jstourac — APPROVED** [review]

**Comment by jstourac** [comment]
This needs to be rebased.

**Comment by kabir** [comment]
@stuartwdouglas needs a rebase, also some stuff in the analysis document

**Comment by stuartwdouglas** [comment]
@kabir done

File: undertow/src/test/resources/org/wildfly/extension/undertow/undertow-reject.xml [hunk 1]
```diff
@@ -31,7 +31,7 @@
       <https-listener disallowed-methods="" max-buffered-request-size="50000"  allow-unescaped-characters-in-url="true"  max-connections="100" name="https" record-request-start-time="true" require-host-http11="true" resolve-peer-address="true" security-realm="UndertowRealm" socket-binding="https-non-default" verify-client="REQUESTED"/>
       <https-listener certificate-forwarding="true" enabled-cipher-suites="ALL:!MD5:!DHA" enabled-protocols="SSLv3, TLSv1.2" name="https-2" proxy-address-forwarding="true" read-timeout="-1" security-realm="UndertowRealm" socket-binding="https-2" write-timeout="-1"/>
       <https-listener disallowed-methods="" max-buffered-request-size="50000" max-connections="100" name="https-4" record-request-start-time="true" resolve-peer-address="true" socket-binding="https-4" security-realm="UndertowRealm"/>
-      <host alias="localhost,some.host" default-response-code="503" default-web-module="something.war" name="default-virtual-host">
+      <host alias="localhost,some.host" default-response-code="503" default-web-module="something.war" name="default-virtual-host" >
```
Comment (kabir) [inline undertow/src/test/resources/org/wildfly/extension/undertow/undertow-reject.xml]: This needs to have queue-requests-on-start="true", and the tests should be adjusted

**Review by kabir — DISMISSED** [review]
@stuartwdouglas  Going through this in a bit more detail, I see that we are missing the rejection tests. Also, apart from the UndertowTransformersTestCase.testConvertTransformersEAP_7_1_0() (which I believe is testing a special case) we should have standard + reject transformer tests against 7.1.0.

**Pull request merged**